Modules/_testcapimodule.c

```diff
@@ -5932,21 +5932,79 @@ test_code_api(PyObject *self, PyObject *Py_UNUSED(args))
     if (co == NULL) {
         return NULL;
     }
-    PyObject *co_code = PyCode_GetCode(co);
-    if (co_code == NULL) {
-        Py_DECREF(co);
-        return NULL;
-    }
-    assert(PyBytes_CheckExact(co_code));
-    if (PyObject_Length(co_code) == 0) {
-        PyErr_SetString(PyExc_ValueError, "empty co_code");
-        Py_DECREF(co);
+    /* co_code */
+    {
+        PyObject *co_code = PyCode_GetCode(co);
+        if (co_code == NULL) {
+            goto fail;
+        }
+        assert(PyBytes_CheckExact(co_code));
+        if (PyObject_Length(co_code) == 0) {
+            PyErr_SetString(PyExc_ValueError, "empty co_code");
+            Py_DECREF(co_code);
+            goto fail;
+        }
         Py_DECREF(co_code);
-        return NULL;
+    }
+    /* co_varnames */
+    {
```

Comment:
Would it make more sense to expose the functions here, and test them in `test/test_capi.py`, checking that the C version returns the same as the Python version?

Reply:
It might be better, but I'm fine with these tests.